web/src/shared/components/pedigree/TangledTree.tsx

```diff
@@ -1,29 +1,774 @@
 /* Inspired by https://observablehq.com/@nitaku/tangled-tree-visualization-ii */
```

Comment:
I haven't reviewed this file in too much detail but happy for it to go in as-is. My feeling is that it could probably be simplified by moving away from SVG and relying on the DOM layout engine to organize the nodes but I also suspect I'm underestimating the complexity. Will have a bit of a better look at it at a later date.